File: wear/src/main/java/hibernate/v2/testyourandroid/ui/fragment/TestColorFragment.java
package hibernate.v2.testyourandroid.ui.fragment;

import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;

import butterknife.BindView;
import butterknife.ButterKnife;
import hibernate.v2.testyourandroid.R;

/**
 * Created by himphen on 21/5/16.
 */
public class TestColorFragment extends BaseFragment {

	@BindView(R.id.colorView)
	View colorView;

	private int i = 0;

	public TestColorFragment() {
		// Required empty public constructor
	}

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
	                         Bundle savedInstanceState) {
		View rootView = inflater.inflate(R.layout.fragment_color, container, false);
		ButterKnife.bind(this, rootView);
		return rootView;
	}

	@Override
	public void onViewCreated(View view, Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
		init();
		mContext.getWindow().setFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON,
				WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
	}

	private void changeColor(int j) {
		switch (j) {
			case 0:
				colorView.setBackgroundColor(Color.RED);
				break;
			case 1:
				colorView.setBackgroundColor(Color.GREEN);
				break;
			case 2:
				colorView.setBackgroundColor(Color.BLUE);
				break;
			case 3:
				colorView.setBackgroundColor(Color.CYAN);
				break;
			case 4:
				colorView.setBackgroundColor(Color.MAGENTA);
				break;
			case 5:
				colorView.setBackgroundColor(Color.YELLOW);
				break;
			case 6:
				colorView.setBackgroundColor(Color.BLACK);
				break;
			case 7:
				colorView.setBackgroundColor(Color.WHITE);
				break;
			case 8:
				colorView.setBackgroundColor(Color.GRAY);
				break;
			case 9:
				colorView.setBackgroundColor(Color.DKGRAY);
				break;
			case 10:
				colorView.setBackgroundColor(Color.LTGRAY);
				i = -1;
				break;
		}
	}

	private void init() {
		colorView.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View view) {
				i++;
				changeColor(i);
			}
		});
	}
}

File: wear/src/main/java/hibernate/v2/testyourwear/ui/test/TestColorFragment.kt
package hibernate.v2.testyourwear.ui.test

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import hibernate.v2.testyourwear.databinding.FragmentColorBinding
import hibernate.v2.testyourwear.ui.base.BaseFragment

/**
 * Created by himphen on 21/5/16.
 */
class TestColorFragment : BaseFragment<FragmentColorBinding>() {

    override fun getViewBinding(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): FragmentColorBinding =
        FragmentColorBinding.inflate(inflater, container, false)

    private var i = 0

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        viewBinding!!.colorView.setOnClickListener {
            changeColor(++i)
        }

        activity?.window?.setFlags(
            WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON,
            WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
        )
    }

    private fun changeColor(j: Int) {
        when (j) {
            0 -> viewBinding?.colorView?.setBackgroundColor(Color.RED)
            1 -> viewBinding?.colorView?.setBackgroundColor(Color.GREEN)
            2 -> viewBinding?.colorView?.setBackgroundColor(Color.BLUE)
            3 -> viewBinding?.colorView?.setBackgroundColor(Color.CYAN)
            4 -> viewBinding?.colorView?.setBackgroundColor(Color.MAGENTA)
            5 -> viewBinding?.colorView?.setBackgroundColor(Color.YELLOW)
            6 -> viewBinding?.colorView?.setBackgroundColor(Color.BLACK)
            7 -> viewBinding?.colorView?.setBackgroundColor(Color.WHITE)
            8 -> viewBinding?.colorView?.setBackgroundColor(Color.GRAY)
            9 -> viewBinding?.colorView?.setBackgroundColor(Color.DKGRAY)
            10 -> {
                viewBinding?.colorView?.setBackgroundColor(Color.LTGRAY)
                i = -1
            }
        }
    }
}